test/clj_gatling/simulation_test.clj

```diff
@@ -668,26 +662,29 @@
                          :weight 1
                          :steps [(step "step1" true)]}
         result (group-by :name
-                         (run-two-scenarios main-scenario second-scenario :concurrency 10 :requests 100))
+                         (run-two-scenarios main-scenario second-scenario :concurrency 10 :requests 10000))
         count-requests (fn [name] (reduce + (map #(count (:requests %)) (get result name))))]
-    (is (approximately== (count-requests "Main") 66 :accuracy 20))
-    (is (approximately== (count-requests "Second") 33 :accuracy 20))
-    (is (approximately== 100 (+ (count-requests "Main") (count-requests "Second")) :accuracy 5))))
+    (is (approximately== (count-requests "Main") 6666 :accuracy 35))
+    (is (approximately== (count-requests "Second") 3333 :accuracy 35))
+    (is (approximately== (+ (count-requests "Main") (count-requests "Second")) 10000 :accuracy 5))))
 
 (deftest after-force-stop-fn-is-called-new-scenarios-are-not-started-anymore
   (let [sent-requests-when-force-stop-requested (atom 0)
         force-stopping-tracker (fn [{:keys [force-stop-fn
                                             sent-requests]}]
-                                 (reset! sent-requests-when-force-stop-requested sent-requests)
-                                 (force-stop-fn))
+                                 (force-stop-fn)
+                                 (reset! sent-requests-when-force-stop-requested sent-requests))
         results (run-single-scenario {:name "progress-tracker-scenario"
-                                      :steps [(step "step" true)]}
+                                      :steps [(step "step" true 10)]}
                                      :concurrency 1
-                                     :requests 500
+                                     :requests 1000
                                      :progress-tracker force-stopping-tracker)
         request-count (count (map :requests results))]
-    (is (< request-count 500))
-    (is (= @sent-requests-when-force-stop-requested request-count))))
+    (is (< request-count 1000))
+    ;; There is a race condition between calling the force-stop-fn and sending
+    ;; more requests, and test requests are very fast, so this will be a little
+    ;; out
+    (is (= request-count @sent-requests-when-force-stop-requested))))
```

Comment:
As with `core_test`, most of the changes here are due to the raw numbers being more approximate now. We can't guarantee that we will have exactly `n` results, but we do know what every result should look like regardless of how many there are. So test them all for that, and also test for an approximate number of them where that is a valid thing to check.